pkg/loop/internal/relayer/pluginprovider/chainreader/chain_reader_test.go

```diff
@@ -287,6 +288,18 @@ func (it *fakeChainReaderInterfaceTester) GetChainReader(_ *testing.T) types.Con
 	return it.impl
 }
 
+func (it *fakeChainReaderInterfaceTester) Start(t *testing.T) {
+	fake, ok := it.impl.(*fakeChainReader)
+	assert.True(t, ok)
+	require.NoError(t, fake.Start(context.Background()))
+}
+
+func (it *fakeChainReaderInterfaceTester) Close(t *testing.T) {
+	fake, ok := it.impl.(*fakeChainReader)
+	assert.True(t, ok)
+	require.NoError(t, fake.Close())
+}
```

Comment:
Why do we need these, why not directly call Start and Close on the ChainReader returned from GetChainReader

Reply:
It fails when we run the LOOPs tests. It tries to execute [this](https://github.com/smartcontractkit/chainlink-common/blob/v0.2.1/pkg/loop/internal/goplugin/service.go#L34-L35) `Start()` func which is a no-op. Exposing these 2 funcs in the `InterfaceTester` was a workaround on this issue.

Let me know your thoughts if there's other way of achieving this without exposing these methods. I'm not very familiar with LOOPs yet and might be missing some context

Reply:
Casting the interface to a specific struct type does not affect which method is called by name via `Start`/`Close`. Please see my other comment: https://github.com/smartcontractkit/chainlink-common/pull/705#discussion_r1758735194